src/main/java/com/spotify/reaper/service/AutoSchedulingManager.java

```diff
@@ -0,0 +1,60 @@
+package com.spotify.reaper.service;
+
+import com.spotify.reaper.AppContext;
+import com.spotify.reaper.core.Cluster;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class AutoSchedulingManager extends TimerTask {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AutoSchedulingManager.class);
+  private static AutoSchedulingManager repairAutoSchedulingManager;
+
+  private final AppContext context;
+  private final ClusterRepairScheduler clusterRepairScheduler;
+
+  public static synchronized void start(AppContext context) {
+    if (null == repairAutoSchedulingManager) {
+      LOG.info("Starting new {} instance. First check in {}ms. Subsequent polls every {}ms",
+          AutoSchedulingManager.class.getSimpleName(),
+          context.config.getAutoScheduling().getInitialDelayPeriod().toMillis(),
+          context.config.getAutoScheduling().getPeriodBetweenPolls().toMillis()
+      );
+
+      repairAutoSchedulingManager = new AutoSchedulingManager(context);
+      Timer timer = new Timer("AutoSchedulingManagerTimer");
+      timer.schedule(
+          repairAutoSchedulingManager,
+          context.config.getAutoScheduling().getInitialDelayPeriod().toMillis(),
+          context.config.getAutoScheduling().getPeriodBetweenPolls().toMillis()
+      );
+    } else {
+      LOG.warn("there is already one instance of {} running, not starting new one", AutoSchedulingManager.class.getSimpleName());
+    }
+  }
+
+  private AutoSchedulingManager(AppContext context) {
+    this(context, new ClusterRepairScheduler(context));
+  }
+
+  public AutoSchedulingManager(AppContext context, ClusterRepairScheduler clusterRepairScheduler) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1213)